src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -92,27 +92,55 @@ private static IEnumerable<string> GetDotNetBasePaths(string workingDirectory)
             bool isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
             string exeName = isWindows ? "dotnet.exe" : "dotnet";
 
-            // We will generally find the dotnet exe on the path, but on linux, it is often just a 'dotnet' symlink (possibly even to more symlinks) that we have to resolve
-            // to the real dotnet executable.
-            // This will work as often as just invoking dotnet from the command line, but we can be more confident in finding a dotnet executable by following
-            // https://github.com/dotnet/designs/blob/main/accepted/2021/install-location-per-architecture.md
-            // This can be done using the nethost library. We didn't do this previously, so I did not implement this extension.
-            foreach (string dir in Environment.GetEnvironmentVariable("PATH").Split(Path.PathSeparator))
-            {
-                string filePath = Path.Combine(dir, exeName);
-                if (File.Exists(filePath))
+            // First check for the DOTNET_ROOT environment variable, as it's often there as with, for example, dotnet format.
+            string dotnet_root = Environment.GetEnvironmentVariable("DOTNET_ROOT");
+            if (!string.IsNullOrEmpty(dotnet_root))
+            {
+                string fullPathToDotnetFromRoot = Path.Combine(dotnet_root, exeName);
+                if (File.Exists(fullPathToDotnetFromRoot))
                 {
-                    if (!isWindows)
+                    dotnetPath = fullPathToDotnetFromRoot;
+                }
+            }
+
+            // Second, check for the DOTNET_ROOT(x86) environment variable, as it can be there, too.
+            if (dotnetPath is null)
```

Comment:
Should this be filtered to only if we are currently in an x86 process? It feels weird to respect `DOTNET_ROOT(x86)` in an amd64/x64 process.